addons/portal/wizard/portal_share.py

```diff
@@ -44,16 +44,19 @@ def _compute_access_warning(self):
                     record = res_model.browse(rec.res_id)
                     rec.access_warning = record.access_warning
 
-    def action_send_mail(self):
-        active_record = self.env[self.res_model].browse(self.res_id)
-        note = self.env.ref('mail.mt_note')
-        signup_enabled = self.env['ir.config_parameter'].sudo().get_param('auth_signup.invitation_scope') == 'b2c'
+    @api.model
+    def _get_note(self):
+        return self.env.ref('mail.mt_note')
 
-        if hasattr(active_record, 'access_token') and active_record.access_token or not signup_enabled:
-            partner_ids = self.partner_ids
-        else:
-            partner_ids = self.partner_ids.filtered(lambda x: x.user_ids)
-        # if partner already user or record has access token send common link in batch to all user
+    def _get_active_record(self):
+        self.ensure_one()
+        return self.env[self.res_model].browse(self.res_id)
+
+    def _send_public_link(self, active_record, note, partners=None):
+        if not note:
+            note = self._get_note()
+        if not partners:
+            partners = self.partner_ids
         for partner in self.partner_ids:
```

Comment:
So we can pass in a list of partners, and if we don't then it's the partner_ids, but then we ignore it and just iterate the partner_ids anyway?

Reply:
Indeed, sorry... I also had to change the for loop like this:
```diff
-for partner in self.partner_ids:
+for partner in partners:
```